api/src/main/java/org/commonjava/indy/data/StoreValidatorRemote.java

```diff
@@ -0,0 +1,162 @@
+package org.commonjava.indy.data;
+
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpHead;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.commonjava.indy.model.core.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.*;
+
+
+@RemoteStoreValidator
+public class StoreValidatorRemote implements StoreValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StoreValidatorRemote.class);
+    private final CountDownLatch httpRequestsLatch = new CountDownLatch(2);
+    private final ExecutorService executorService = Executors.newFixedThreadPool(2);
+    private static final String REMOTE = "remote";
+
+
+    @Override
+    public ArtifactStoreValidateData validate(ArtifactStore artifactStore)
+        throws InvalidArtifactStoreException, MalformedURLException {
+        // Parse Remote Repository for checking if it is a valid URL
+        try {
+            //#region
+//            LOGGER.info("=> ArtifactStore Type: " + artifactStore.getType().singularEndpointName());
+            if (REMOTE.equalsIgnoreCase(artifactStore.getType().singularEndpointName())
+                || artifactStore instanceof RemoteRepository) {
+                // Cast to Remote Repository
+                RemoteRepository remoteRepository = (RemoteRepository) artifactStore;
+                //Validate URL from remote Repository URL
+                Optional<URL> remoteUrl = Optional.of(new URL(remoteRepository.getUrl()));
+                // Execute HTTP GET & HEAD requests in separate thread pool from executor service
+                Future<Boolean> httpGetStatus = executorService.submit(new HttpGetTask(new HttpGet(remoteUrl.get().toURI())));
```

Comment:
I'd suggest using a completion service (we have one in Weft that makes it easy to drain, or consume completed tasks, easily)...but in this case, the latch makes sense since we have only two requests to handle.